core/tee/tee_svc.c

```diff
@@ -845,13 +862,24 @@ static TEE_Result tee_svc_update_out_param(
 						return res;
 				}
 			}
-			usr_param->vals[n * 2 + 1] = sz;
+			res = PUT_USER_SCALAR(sz, &usr_param->vals[n * 2 + 1]);
+			if (res)
+				return res;
+
 			break;
 
 		case TEE_PARAM_TYPE_VALUE_OUTPUT:
 		case TEE_PARAM_TYPE_VALUE_INOUT:
-			vals[n * 2] = param->u[n].val.a;
-			vals[n * 2 + 1] = param->u[n].val.b;
+			a = (uint64_t)param->u[n].val.a;
+			b = (uint64_t)param->u[n].val.b;
+
+			res = PUT_USER_SCALAR(a, &vals[n * 2]);
```

Comment:
How about using an array of 2 uint64_t and using a single `copy_to_user()`?

Reply:
sgtm. Updated